### PR TITLE
 Query By Filename

### LINKED ISSUE #59: Data Access API Add Query Option for Filename
### Summary

Within the Query code, add an option to search for a file by the filename. See User Testing #40 for more information.

### Requirements

The more specific the better.
  - [ ]
  - [ ]

### Related tickets

_No response_

### PR DESCRIPTION
**Pull request by daralynnrhode**
# Adding a Query by filename argument

## Overview

Giving the option to query to see if a certain file based off of a given file path is present. Users would pass in `query --filename [File]` and the file will either be returned or no matching files will be returned. 
To avoid overwriting other passed in parameters, the filename must be the only argument passed in to the query.

Closes #59 
## Updated Files

- cli.py
   - added `--filename` to acceptable arguments
   - Code to pass the given file information into the query

### CODE REVIEW

File: imap_data_access/cli.py
Comment (maxinelasp) [inline imap_data_access/cli.py:121]: To address your doubts - this seems ok to me. You could also check to see if "instrument", "data_level", etc is in sys.argv, which would be more reliable in case we add some additional options which _can_ be passed in alongside filename.

Comment (maxinelasp) [reply imap_data_access/cli.py:121]: Actually, you could also validate that `query_params` is empty, since we would expect it to be empty if `filename` was passed in, and we're rewriting it anyway so it's a good thing to check

**Review by maxinelasp — COMMENTED** [review]
Looks like this PR does include commits from the other change to add "latest" - which is fine, but don't merge those changes until the other PR is approved (or remove them from here)

**Review by maxinelasp — APPROVED** [review]
Looks good!

File: imap_data_access/cli.py
Comment (bourque) [inline imap_data_access/cli.py:309]: Not absolutely necessary, but might be nice to add a link to the naming convention documentation in the help string here, so users can find what the file naming convention is in case they don't know it:

https://imap-processing.readthedocs.io/en/latest/development-guide/style-guide/naming-conventions.html#data-product-file-naming-conventions

**Review by bourque — APPROVED** [review]
LGTM! Nice work.

**Pull request merged**
